src/pybind/mgr/dashboard/frontend/src/app/shared/directives/autofocus.directive.ts

```diff
@@ -1,17 +1,24 @@
-import { AfterViewInit, Directive, ElementRef } from '@angular/core';
+import { AfterViewInit, Directive, ElementRef, Input } from '@angular/core';
 
 import * as _ from 'lodash';
 
 @Directive({
   selector: '[autofocus]' // tslint:disable-line
 })
 export class AutofocusDirective implements AfterViewInit {
+  private focus = false;
+
   constructor(private elementRef: ElementRef) {}
 
   ngAfterViewInit() {
     const el: HTMLInputElement = this.elementRef.nativeElement;
-    if (_.isFunction(el.focus)) {
+    if (this.focus && _.isFunction(el.focus)) {
       el.focus();
     }
   }
+
+  @Input()
+  public set autofocus(condition: any) {
+    this.focus = condition !== false;
```

Comment:
How about using a boolean or function if delivered else use the focus like before?

```suggestion
    if (_.isBoolean(focus) {
      this.focus = focus
    } else if (_.isFunction(focus)) {
      this.focus = focus()
    }
```
and on line 9 change `focus` to true.

Reply:
Done